Update docs

GeorgianaElena opened the pull request:
This revamps the docs a bit. I didn't manage to cover everything, but I reordered the sections a bit, added the Jupyter logo and the nice badges from team-compass.

Closes https://github.com/jupyterhub/oauthenticator/issues/332

consideRatio commented on README.md:
```diff
@@ -1,25 +1,56 @@
+* [Okpy](https://oauthenticator.readthedocs.io/en/latest/getting-started.html#okpyauthenticator)
+* [Globus](https://oauthenticator.readthedocs.io/en/latest/getting-started.html#globus-setup)
+* [Moodle](https://oauthenticator.readthedocs.io/en/latest/getting-started.html#moodle-setup)
+* [Yandex](https://oauthenticator.readthedocs.io/en/latest/getting-started.html#yandex-setup)
```
Woops the Yandex authenticator is removed I think, so the documentation should also be removed.

GeorgianaElena replied:
I know, but the example setup for Yandex is using GenericAuthenticator :yellow_heart:

GeorgianaElena replied:
And I added a similar example for AWSCognito. I didn't tested it though :eyes:

consideRatio replied:
Ah!

consideRatio commented:
:heart: wooooo excellent work! This has been one of my "oh this really has to get done" mental notes I've had for a while!

consideRatio commented on README.md:
```diff
@@ -1,25 +1,56 @@
+[OpenShift](oauthenticator/openshift.py).
+
+There is also a [GenericAuthenticator](oauthenticator/generic.py)
+that can be used to write your own Authenticators with any OAuth 2.0 provider.
```
> can be used to write your own Authenticators with any OAuth 2.0 provider.

This sounds to me like we can only use the generic authenticator to create a custom class from it, but it can also be used on its own through configuration with any OAuth 2.0 provider. Sometimes though, we need something additional, and then a custom class is needed that can use the generic authenticator as a base class.

consideRatio commented on docs/source/getting-started.rst:
```diff
@@ -1,65 +1,30 @@
 2. register with your provider
 3. configure JupyterHub (if supported, pick oauthenticator class, otherwise see :doc:`writing-an-oauthenticator`).
 4. common configuration (client_id, client_secret, callback_url, whitelist, etc.)
 5. specific configuration for your provider
```
- I think it would make sense to say `identity provider` throughout instead of just provider.
- I think point 3 is a bit too vague, it could be choose authenticator class or something like that instead, and avoid suggesting the creation of a custom authenticator before falling back to the generic authenticator first, and only if that isn't enough, consider creating your own. I think, not suggesting to create an own at all is best here but instead suggesting to choose a specific or the generic authenticator class.
- "common configuration" could be reworded as "configure the authenticator class" or similar to assume less from the reader.

consideRatio commented on docs/source/getting-started.rst:
```diff
@@ -1,65 +1,30 @@
-
-For an example docker image using OAuthenticator, see the
-`examples <examples>`__ directory.
+OAuthenticator currently supports the following **authentication services**:
```
Here "authentication services" are used, hmmm.. Since "provider" (which is referring to "identity provider") has been used before, I like to use that here as well.

consideRatio commented on docs/source/getting-started.rst:
```diff
@@ -1,65 +1,30 @@
-   cd oauthenticator
-   pip3 install -e .
+A `generic implementation <oauthenticator.generic.GenericOAuthenticator>`__, which you can
+use with **any provider**, is also available.
```
any "OAuth2 identity provider"

GeorgianaElena commented:
Thanks for the feedback @consideRatio :heart:. I've just added a new commit that addresses your comments.

I know the test failures are unrelated. I think it has something to do with the [`force_instace = True`](https://github.com/jupyterhub/oauthenticator/blob/master/oauthenticator/generic.py#L74) just added but I don't know why. Do you have any ideas on how to fix them?

consideRatio commented:
Sorry for slow followup @GeorgianaElena, thanks so much for putting in this work! It makes all the difference to have good documentation! :heart: :tada: I'm very happy about this PR!!
